src/edgepi/adc/edgepi_adc.py

```diff
@@ -641,12 +640,14 @@ def reset(self):
         self.transfer(self.adc_ops.reset_adc())
         self.__reapply_config()
 
-    def __is_data_ready(self):
+    def __is_data_ready(self, adc_num: ADCNum):
         # pylint: disable=unused-private-member
         # required for integration testing in test_conversion_times.py
         """Utility for testing conversion times, returns True if ADC indicates new voltage data"""
-        read_data = self.transfer([ADCComs.COM_RDATA1.value] + [255] * 6)
-        return read_data[1] & 0b01000000
+        read_data = self.transfer([adc_num.value.read_cmd] + [255] * 6)
+        _logger.debug(f"__is_data_ready: cmd {adc_num.value.read_cmd,} read_data {read_data[1]}")
+        return (read_data[1] & 0b01000000) == 0b01000000 if adc_num is ADCNum.ADC_1 else\
+               (read_data[1] & 0b10000000) == 0b10000000
```

Comment:
you remember "explicit is better than implicit". I would add another if for ADC_2

Reply:
540467f61a29a8292b1502dca2736f9992d277a9

Reply:
how about:
```
        if adc_num is ADCNum.ADC_1:
            return (read_data[1] & 0b01000000) == 0b01000000

        if adc_num is ADCNum.ADC_2:
            return (read_data[1] & 0b10000000) == 0b10000000

        raise ValueError(f'Invalid adc_num={adc_num}')
```

Reply:
https://github.com/EdgePi-Cloud/edgepi-python-sdk/blob/b61170324d76e0e88aa57f1b27a1958fabb55601/src/edgepi/adc/edgepi_adc.py#L647-L649

invalid adc_num will throw an error on line 647. It's supposed to be an enum with a dataclass as a value. It will throw an attribute error or type error